library/src/main/java/com/google/android/exoplayer/util/ManifestFetcher.java

```diff
@@ -333,7 +348,6 @@ public void onLoadCompleted(Loadable loadable) {
 
     @Override
     public void onLoadCanceled(Loadable loadable) {
-      // This shouldn't ever happen, but handle it anyway.
```

Comment:
I don't think onSingleManifestError should be invoked if cancellation is now expected, so this would need to do something else.